zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/ExecutionListenerJobTest.java

```diff
@@ -0,0 +1,640 @@
+          .startEvent()
+          .serviceTask("task", t -> t.zeebeJobType("service_task_type"))
+          .zeebeStartExecutionListener("start_EL_type")
+          .endEvent()
+          .done();
+
+  @Rule
+  public final RecordingExporterTestWatcher recordingExporterTestWatcher =
+      new RecordingExporterTestWatcher();
+
+  @Test
+  public void shouldCompleteServiceTaskWithExecutionListeners() {
+    // given
+    final BpmnModelInstance modelInstance =
+        Bpmn.createExecutableProcess(PROCESS_ID)
+            .startEvent()
+            .serviceTask("task", t -> t.zeebeJobType("service_task_type"))
+            .zeebeStartExecutionListener("start_EL_type_1")
+            .zeebeStartExecutionListener("start_EL_type_2")
+            .zeebeEndExecutionListener("end_EL_type")
+            .endEvent()
+            .done();
+    // when
+    ENGINE.deployment().withXmlResource(modelInstance).deploy();
+    final long processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    // then
+    assertJobLifecycle(processInstanceKey, 0, "start_EL_type_1", JobKind.EXECUTION_LISTENER);
+    assertJobLifecycle(processInstanceKey, 1, "start_EL_type_2", JobKind.EXECUTION_LISTENER);
+    assertJobLifecycle(processInstanceKey, 2, "service_task_type", JobKind.BPMN_ELEMENT);
+    assertJobLifecycle(processInstanceKey, 3, "end_EL_type", JobKind.EXECUTION_LISTENER);
+
+    final ProcessInstanceRecordStream processInstanceRecordStream =
+        RecordingExporter.processInstanceRecords()
+            .withProcessInstanceKey(processInstanceKey)
+            .limitToProcessInstanceCompleted();
+    assertThat(processInstanceRecordStream)
+        .extracting(r -> r.getValue().getBpmnElementType(), Record::getIntent)
+        .containsSubsequence(
+            tuple(BpmnElementType.SERVICE_TASK, ProcessInstanceIntent.ELEMENT_ACTIVATING),
+            tuple(BpmnElementType.SERVICE_TASK, ProcessInstanceIntent.EXECUTION_LISTENER_COMPLETED),
+            tuple(BpmnElementType.SERVICE_TASK, ProcessInstanceIntent.EXECUTION_LISTENER_COMPLETED),
+            tuple(BpmnElementType.SERVICE_TASK, ProcessInstanceIntent.ELEMENT_ACTIVATED),
+            tuple(BpmnElementType.SERVICE_TASK, ProcessInstanceIntent.COMPLETE_ELEMENT),
+            tuple(BpmnElementType.SERVICE_TASK, ProcessInstanceIntent.ELEMENT_COMPLETING),
+            tuple(BpmnElementType.SERVICE_TASK, ProcessInstanceIntent.EXECUTION_LISTENER_COMPLETED),
+            tuple(BpmnElementType.SERVICE_TASK, ProcessInstanceIntent.ELEMENT_COMPLETED),
+            tuple(BpmnElementType.PROCESS, ProcessInstanceIntent.ELEMENT_COMPLETED));
+  }
+
+  @Test
+  public void shouldRetryExecutionListener() {
+    // given
+    ENGINE.deployment().withXmlResource(SIMPLE_PROCESS).deploy();
+    final long processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    // when
+    ENGINE.job().ofInstance(processInstanceKey).withType("start_EL_type").withRetries(1).fail();
```

Comment:
:wrench: We should extract the EL job type as a constant. The job type is defined in the static model and used inside the test cases.

Reply:
Applied

Reply:
Good point!
I've revisited all tests and aligned the naming for service task/EL types.
Thanks for highlighting this!